webapp/src/components/backend_selector.jsx

```diff
@@ -0,0 +1,137 @@
+        value: PropTypes.oneOfType([
+            PropTypes.object,
+            PropTypes.array,
+            PropTypes.string,
+        ]),
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            options: [],
+            invalid: false,
+            error: '',
+        };
+    }
+
+    componentDidMount() {
+        if (this.props.addValidate) {
+            this.props.addValidate(this.isValid);
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.props.removeValidate) {
+            this.props.removeValidate(this.isValid);
+        }
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (prevState.invalid && this.props.value !== prevProps.value) {
+            this.setState({invalid: false}); //eslint-disable-line react/no-did-update-set-state
+        }
+    }
+
+    onChange = (options) => {
+        if (!options) {
+            this.props.onChange(this.props.isMulti ? [] : '');
+            return;
+        }
+
+        this.props.onChange(this.props.isMulti ? options.map((v) => v.value) : options.value);
+
+        if (this.props.resetInvalidOnChange) {
+            this.setState({invalid: false});
+        }
+    };
+
+    isValid = () => {
+        if (!this.props.required) {
+            return true;
+        }
+
+        const valid = Boolean(this.props.value && this.props.value.toString().length !== 0);
+        this.setState({invalid: !valid});
+        return valid;
+    };
+
+    errorComponent = () => {
+        if (!this.state.error) {
+            return null;
+        }
+
+        return (
+            <p className='alert alert-danger'>
+                <i
+                    className='fa fa-warning'
+                    title='Warning Icon'
+                />
+                <span> {this.state.error.toString()}</span>
+            </p>
+        );
+    };
+
+    validationError = () => {
+        if (!this.props.required || !this.state.invalid) {
+            return null;
+        }
+
+        return (
+            <p className='help-text error-text'>
+                <span>{'This field is required.'}</span>
+            </p>
+        );
+    };
+
+    loadOptions = async () => {
+        const options = await this.props.load();
+
+        // prevent re-render if the options remain unchanged
+        if (JSON.stringify(options) === JSON.stringify(this.state.options)) {
+            return;
+        }
+
+        this.setState({options});
+    }
+
+    render = () => {
+        this.loadOptions();
+
+        return (
+            <Setting {...this.props}>
+                <ReactSelect
+                    required={this.props.required}
+                    isMulti={this.props.isMulti}
+                    name={this.props.name}
+                    onChange={this.onChange}
+                    options={this.state.options}
+                    styles={getStyleForReactSelect(this.props.theme)}
+                />
```

Comment:
Since five of these props are passed in from the parent, we can shorthand them as `...this.props`
```suggestion
                <ReactSelect
                    {...this.props}
                    styles={getStyleForReactSelect(this.props.theme)}
                />
```